pkg/api/query/v1.go

```diff
@@ -1301,6 +1311,54 @@ func (qapi *QueryAPI) stores(_ *http.Request) (interface{}, []error, *api.ApiErr
 	return statuses, nil, nil, func() {}
 }
 
+func (qapi *QueryAPI) getLabelMatchers(matchers []string, tenant string) ([][]*labels.Matcher, *api.ApiError) {
```

Comment:
If we use the `EnforceQueryTenancy` middleware to do the label injection, we don't the label injection logic also here.